core/src/main/java/software/amazon/awssdk/core/client/builder/AsyncClientBuilder.java

```diff
@@ -49,4 +50,20 @@
      * finished with it, the SDK will only close HTTP clients that it creates.
      */
     B asyncHttpConfiguration(ClientAsyncHttpConfiguration asyncHttpConfiguration);
+
+    /**
+     * Configures the HTTP client used by the service client.
+     *
+     * Allows just specifying the builder parameters without having to instantiate a builder and call <code>build</code> on it.
+     * For example
+     *
+     * <pre><code>
+     *     builder().asyncHttpConfiguration(b -> b.httpClient(client));
+     * </code></pre>
+     *
+     * @see #asyncHttpConfiguration(ClientAsyncHttpConfiguration)
+     */
+    default B asyncHttpConfiguration(UnaryOperator<ClientAsyncHttpConfiguration.Builder> asyncHttpConfiguration) {
```

Comment:
Should this be a `Consumer` to match builder's `apply`?

Reply:
yup - could be - I never liked that it mutated it, but that's the pattern we've followed before so it's probably better to remain consistent.